### ISSUE TITLE
Assign to let/const variable in generic proc

### ISSUE THREAD

**Issue by konsumlamm** [nim-lang/Nim#16629]
It is possible to assign values to variables declared as `let` or `const` in generic procs (only an error when instantiated).

### Example
```nim
proc test[T](x: T) =
  let a = 5 # same with const
  a = 42
```

### Current Output
Compiles without problems.

### Expected Output
Error message:
```
Error: 'a' cannot be assigned to
```

### Additional Information
```
$ nim -v
Nim Compiler Version 1.5.1
git hash: 2c2baa9fad1d144c42baf30ed004decdf8bb0483
```

**Comment by Araq** [comment]
See also https://nim-lang.org/docs/manual.html#generics-symbol-lookup-in-generics, most checking of generics is delayed until instantiation.

**Comment by konsumlamm** [comment]
Is there a situation where this behaviour is desirable, i.e. is there a case where instantiation will actually succeed? If not, I don't see a good reason to allow this behaviour.

**Comment by Araq** [comment]
True, but a little besides the point, there are plenty of things that could be eagerly checking inside a generic body, but are not.

**Comment by konsumlamm** [comment]
Could we at least add a warning for this? Since there is no situation were the instantiated code would be valid, I think this is appropriate.

**Comment by Araq** [comment]
There is research work going on behind the scenes in how much we can really typecheck generic code, if fruitful, this would be solved as well. This particular instance however remains to be of low importance as you're supposed to check your generic code.
